cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/analyzer/GoalOptimizer.java

```diff
@@ -448,7 +468,13 @@ public OptimizerResult optimizations(ClusterModel clusterModel,
       operationProgress.addStep(step);
       LOG.debug("Optimizing goal {}", goal.name());
       long startTimeMs = _time.milliseconds();
-      boolean succeeded = goal.optimize(clusterModel, optimizedGoals, optimizationOptions);
+      boolean succeeded;
+      try {
+        succeeded = goal.optimize(clusterModel, optimizedGoals, optimizationOptions);
+      } finally {
+        provisionResponse.aggregate(goal.provisionResponse());
```

Comment:
Could you briefly explain why this change is needed, and what's the provisionResponse when the try fails?

Reply:
I have modified this in the next commit. Let me know if you have any more questions.